### PR TITLE
feat(ref-lib-gen): adds new module @sap-ux/ui5-library-reference-sub-generator

### PR DESCRIPTION
**Pull request by cianmSAP**
#2472 

 adds new module @sap-ux/ui5-library-reference-sub-generator

### CODE REVIEW

File: packages/ui5-library-reference-sub-generator/src/app/index.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import { prompt } from '@sap-ux/ui5-library-reference-inquirer';
+import { t } from '../utils/i18n';
+import { EventName } from '../telemetryEvents';
+import type { UI5ReferenceLibGenerator } from './types';
+import type { Prompts } from '@sap-devx/yeoman-ui-types';
+import type { ReuseLibConfig } from '@sap-ux/ui5-library-reference-writer';
+import type { YeomanEnvironment } from '@sap-ux/fiori-generator-shared';
+import type { InquirerAdapter, UI5LibraryReferenceAnswers } from '@sap-ux/ui5-library-reference-inquirer';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (cianmSAP) [reply packages/ui5-library-reference-sub-generator/src/app/index.ts]: Thanks @longieirl , I've tested and it's not required. I've removed

**Review by longieirl — DISMISSED** [review]
I havent checked alot of the js files.
The implementation looks good.
Test coverage is good.

File: packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_not_sapux/ui5.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ui5Theme: sap_fiori_3
+server:
+  customMiddleware:
+  - name: fiori-tools-proxy
+    afterMiddleware: compression
+    configuration:
+      ignoreCertError: false # If set to true, certificate errors will be ignored. E.g. self-signed certificates will be accepted
+      backend:
+      - path: /sap
+        url: http://ims.saplabs.com:50017
```
[the remark this answers is not in the document]
Comment (devinea) [reply packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_not_sapux/ui5.yaml:14]: Maybe mock server here?

File: packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2/ui5-local.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - name: themelib_sap_fiori_3
+server:
+  customMiddleware:
+  - name: fiori-tools-proxy
+    afterMiddleware: compression
+    configuration:
+      ignoreCertError: false # If set to true, certificate errors will be ignored. E.g. self-signed certificates will be accepted
+      backend:
+      - path: /sap
+        url: http://ims.saplabs.com:50017
```
Comment (devinea) [inline packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2/ui5-local.yaml:26]: remove

File: packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2/ui5.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ui5Theme: sap_fiori_3
+server:
+  customMiddleware:
+  - name: fiori-tools-proxy
+    afterMiddleware: compression
+    configuration:
+      ignoreCertError: false # If set to true, certificate errors will be ignored. E.g. self-signed certificates will be accepted
+      backend:
+      - path: /sap
+        url: http://ims.saplabs.com:50017
```
Comment (devinea) [inline packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2/ui5.yaml:14]: remove

File: packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2_custom_webapp_path/ui5-local.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    - name: themelib_sap_fiori_3
+server:
+  customMiddleware:
+  - name: fiori-tools-proxy
+    afterMiddleware: compression
+    configuration:
+      ignoreCertError: false # If set to true, certificate errors will be ignored. E.g. self-signed certificates will be accepted
+      backend:
+      - path: /sap
+        url: http://ims.saplabs.com:50017
```
Comment (devinea) [inline packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2_custom_webapp_path/ui5-local.yaml:26]: remove

File: packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2_custom_webapp_path/ui5.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ui5Theme: sap_fiori_3
+server:
+  customMiddleware:
+  - name: fiori-tools-proxy
+    afterMiddleware: compression
+    configuration:
+      ignoreCertError: false # If set to true, certificate errors will be ignored. E.g. self-signed certificates will be accepted
+      backend:
+      - path: /sap
+        url: http://ims.saplabs.com:50017
```
Comment (devinea) [inline packages/ui5-library-reference-sub-generator/test/unit/sample/test_project_lrop_v2_custom_webapp_path/ui5.yaml:14]: remove

**Review by devinea — CHANGES_REQUESTED** [review]

File: .changeset/gorgeous-dogs-watch.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+'@sap-ux/ui5-library-sub-generator': patch
+'@sap-ux/fiori-generator-shared': patch
+'@sap-ux/project-access': patch
```
[the remark this answers is not in the document]
Comment (cianmSAP) [reply .changeset/gorgeous-dogs-watch.md:4]: Changeset is amended

**Review by devinea — APPROVED** [review]
New module setup looks good. 
Code changes look good. 
Good coverage. 
Changeset ✅ 

Thanks for the updates.

**Review by longieirl — APPROVED** [review]
All comments addressed.
Code coverage is good.
Changeset is now good.
Havent tested locally, only reviewed the implementation code.

**Pull request merged**